zulipterminal/ui_tools/views.py

```diff
@@ -271,10 +271,11 @@ def update_streams(self, search_box: Any, new_text: str) -> None:
         # wait for any previously started search to finish to avoid
         # displaying wrong stream list.
         with self.search_lock:
-            streams_display = self.streams_btn_list.copy()
-            for stream in self.streams_btn_list:
-                if not stream.stream_name.lower().startswith(new_text.lower()):
-                    streams_display.remove(stream)
+            streams_display = [
+                stream
+                for stream in self.streams_btn_list.copy()
```

Comment:
As you mentioned the `match_stream` helper function could be used here. In fact, the `match_stream` logic may even benefit stream search.